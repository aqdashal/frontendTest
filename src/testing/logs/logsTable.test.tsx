import React from 'react';
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { MOCK_LOGS, columns } from "../data"
import { MemoryRouter } from 'react-router-dom';
import LogsTable from "../../components/table";

describe('<LogsTable>', () => {

    let handleSorting: jest.Mock;
    let tableData, sortField, sortOrder;
    let tableColumns;

    const setup = () => {
        render(
            <MemoryRouter>
                <LogsTable {...{ data: tableData, columns: tableColumns, sortField, sortOrder, handleSorting }} />
            </MemoryRouter>
        );
    }

    beforeEach(() => {
        tableData = MOCK_LOGS;
        sortField = "";
        sortOrder = "";
        tableColumns = columns;
        handleSorting = jest.fn();
    })

    it("should render initially", () => {
        setup();
    })

    it("it renders LogsTable properly", () => {
        setup();
        expect(screen.getByRole("table"))
    })
})