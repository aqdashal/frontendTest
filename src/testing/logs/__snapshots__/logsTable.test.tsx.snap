// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogsTable> snapshot 1`] = `
<div
  className="mt-4"
>
  <table
    className="w-100"
  >
    <thead>
      <tr>
        <th
          className="default border-bottom"
          onClick={[Function]}
        >
          Log ID
          <span>
            <img
              src="default.png"
            />
          </span>
        </th>
        <th
          className="default border-bottom"
          onClick={[Function]}
        >
          Application Type
          <span>
            <img
              src="default.png"
            />
          </span>
        </th>
        <th
          className="default border-bottom"
          onClick={[Function]}
        >
          Application ID
          <span>
            <img
              src="default.png"
            />
          </span>
        </th>
        <th
          className="default border-bottom"
          onClick={[Function]}
        >
          Action
          <span>
            <img
              src="default.png"
            />
          </span>
        </th>
        <th
          className=" border-bottom"
          onClick={null}
        >
          Action Details
          <span>
            <img />
          </span>
        </th>
        <th
          className="default border-bottom"
          onClick={[Function]}
        >
          Date:Time
          <span>
            <img
              src="default.png"
            />
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="border-bottom"
        >
          906468196730134
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          DARI_REFRESH_TOKEN
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2022-01-31 17:29:00
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          365001413757985
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          DARI_REFRESH_TOKEN
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2022-01-31 17:51:01
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          934448782347868
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          DARI_REFRESH_TOKEN
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2022-01-31 17:51:01
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          805901994400875
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          DARI_REFRESH_TOKEN
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2022-01-31 19:28:02
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          842180396585606
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          DARI_REFRESH_TOKEN
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2022-01-31 17:29:00
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          712087267864057
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          DARI_APP_LOGIN
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2022-01-31 17:17:32
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          562514270265046
        </td>
        <td
          className="border-bottom"
        >
          CERT_TITLE_DEED_PLOT
        </td>
        <td
          className="border-bottom"
        >
          375709440378514
        </td>
        <td
          className="border-bottom"
        >
          INITIATE_APPLICATION
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2021-11-29 18:33:40
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          519234858220197
        </td>
        <td
          className="border-bottom"
        >
          LEASE_REGISTRATION
        </td>
        <td
          className="border-bottom"
        >
          657199561191750
        </td>
        <td
          className="border-bottom"
        >
          INITIATE_APPLICATION
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2021-11-28 10:30:32
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          565125103357125
        </td>
        <td
          className="border-bottom"
        >
          ADD_POA
        </td>
        <td
          className="border-bottom"
        >
          726278576002607
        </td>
        <td
          className="border-bottom"
        >
          INITIATE_APPLICATION
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2021-12-09 12:01:12
        </td>
      </tr>
      <tr>
        <td
          className="border-bottom"
        >
          357096041490197
        </td>
        <td
          className="border-bottom"
        >
          LEASE_REGISTRATION
        </td>
        <td
          className="border-bottom"
        >
          144170790324169
        </td>
        <td
          className="border-bottom"
        >
          SUBMIT_APPLICATION
        </td>
        <td
          className="border-bottom"
        >
          ——
        </td>
        <td
          className="border-bottom"
        >
          2021-12-09 02:23:54
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
